src/game/client/components/chat.cpp

```diff
@@ -1468,50 +1509,104 @@ void CChat::Com_Befriend(CChat *pChatData, const char* pCommand)
 
 
 // CChatCommands methods
-CChat::CChatCommands::CChatCommands(CChatCommand apCommands[], int Count) : m_apCommands(apCommands), m_Count(Count), m_pSelectedCommand(0x0) { }
-CChat::CChatCommands::~CChatCommands() { }
+CChat::CChatCommands::CChatCommands() : m_pSelectedCommand(NULL) { }
+CChat::CChatCommands::~CChatCommands() {
+	m_aCommands.delete_all();
+}
 
 // selection
 void CChat::CChatCommands::Reset()
 {
-	m_pSelectedCommand = 0x0;
+	m_pSelectedCommand = NULL;
+}
+
+void CChat::CChatCommands::AddCommand(const char *pName, const char *pArgsFormat, const char *pHelpText, COMMAND_CALLBACK pfnCallback)
+{
+	if(m_aCommands.size() >= MAX_COMMANDS)
+		return;
+
+	CChatCommand *pCommand = new CChatCommand();
+	pCommand->m_pName = (char *)mem_alloc(sizeof(char) * (str_length(pName) + 1), 1);
+	str_copy(pCommand->m_pName, pName, str_length(pName) + 1);
+
+	pCommand->m_pArgsFormat = (char *)mem_alloc(sizeof(char) * (str_length(pArgsFormat) + 1), 1);
+	str_copy(pCommand->m_pArgsFormat, pArgsFormat, str_length(pArgsFormat) + 1);
+
+	pCommand->m_pHelpText = (char *)mem_alloc(sizeof(char) * (str_length(pName) + 1), 1);
```

Comment:
This should be str_length(pHelpText) (and so your write past the buffer and it crashes). Also I don't see the point of heap allocating everything.